dist/test_tag.py

```diff
@@ -89,56 +107,61 @@
         # Check if line is valid
         if not line:
             # Check if invalid line after START_TAG
-            if is_start == True:
-                print("Error syntax in file " + filename)
-                exit()
+            if in_tag_block == True:
+                print("Syntax Error in file: " + filename)
+                exit(1)
             else:
                 continue
 
         # Check if end of test tag
         if END_TAG in line:
             # END_TAG should not be before START_TAG
-            if is_start == False:
-                print("Error syntax in file " + filename + ". Unexpected tag: " + END_TAG)
-                exit()
+            if in_tag_block == False:
+                print("Syntax Error in file: " + filename + ". Unexpected tag: " + END_TAG)
+                exit(1)
             # END_TAG should not be met before a test tag
             if is_file_ignored == False and is_file_tagged == False:
-                print("Error syntax in file " + filename + ". Missing test tag.")
-                exit()
-            is_end = True
+                print("Syntax Error in file: " + filename + ". Missing test tag.")
+                exit(1)
             nb_valid_files = nb_valid_files + 1
             # Go to next file
             break
 
         # Check if start of test tag
         if START_TAG in line:
             # Only one START_TAG is allowed
-            if is_start == True:
-                print("Error syntax in file " + filename + ". Unexpected tag: " + START_TAG)
-                exit()
-            is_start = True
+            if in_tag_block == True:
+                print("Syntax Error in file: " + filename + ". Unexpected tag: " + START_TAG)
+                exit(1)
+            in_tag_block = True
             continue
 
-        if is_start == True:
+        if in_tag_block == True:
+            tag = line
             # Check if file is ignored
             if is_file_ignored == True:
                 print("Unexpected value in ignored file: " + filename)
-                exit()
-            if line == IGNORE_FILE:
+                exit(1)
+            if tag == IGNORE_FILE:
                 nb_ignored_files = nb_ignored_files + 1
                 is_file_ignored = True
                 continue
-            # Check if current tag is valid
-            if not line in valid_tags:
-                print("Tag is not valid ! Add the new tag to test_tags.ok:\n" + line)
-                exit()
-            else:
-                is_file_tagged = True
+
+            # Validate the tag's correctness.
+            tag_count = validate_tag(tag, filename)
+            is_file_tagged = True
+
+            # Add the test type to the tag if it wasn't already.
```

Comment:
What is this special handling for ? Why didnt we have it before if it is necessary ?

Reply:
Essentially to save tags from being too verbose its assumed that .c files are "data_correctness" test types and ".py" is a "functional_correctness" test type.

Reply:
Assuming that we will never have a Python script that is a data_correctness test and vice versa

Reply:
The logic I'm using here is wrong as a valid tag could be 1, 2 or 3, in length. I don't know if theres a clean way to check this. I'll on it for a bit.

Reply:
Alright I've pushed up a change that does it a bit better.